Docs/Book/GettingStartedWithContributing.md

```diff
@@ -0,0 +1,633 @@
+- Local variable names are usually lower-case, but sometimes use either snake_case or camelCase.
+- We follow the formatting suggestions from [Google's C++ style guidelines](https://google.github.io/styleguide/cppguide.html).
+- Use of an auto-formatter (e.g. [clang-format](http://clang.llvm.org/docs/ClangFormat.html)) is strongly recommended. This can be integrated with editors, but note that at least version 3.8 of clang-format is required as the configuration files are not backwards compatible. There is a .clang-format config file at the root of the repository; code layout should be based on this configuration.
+- The public classes / functions exposed in the API are typically defined in the header file, and the underlying method in the C++ file of the same name.
+- Use multi-line comments in the header file to document code:
+    ```c++
+    /*!
+    this is a documentation string
+    */
+    ```
+
+We would also recommend checking your code for memory leaks with [Valgrind](https://valgrind.org/) before opening a pull request.
+
+### Tests
+
+The RDKit C++ Tests use the [Catch2 framework](https://github.com/catchorg/Catch2). The tests generally are placed in a file named "catch_tests.cpp" in the same directory as the corresponding code. Small changes and bug fixes may only need the addition of a new TEST\_CASE to an existing test set rather than a new file.
+
+For example, to test the atom count function:
+
+```c++
+#include "RDGeneral/test.h"
+#include <catch2 /catch_all.hpp>
+#include <GraphMol /SmilesParse/SmilesParse.h>
+
+using namespace RDKit;
+
+TEST_CASE("test basic Mol features", "[ROMol]") {
+  SECTION("basics") {
+    {
+    // shortcut to create a molecule from SMILES
+    auto m = "CCO"_smiles;
+    // check that the molecule was created successfully
+    REQUIRE(m);
+    // test some functionality
+    int nAtoms = m->getNumAtoms();
+    CHECK(nAtoms == 3);
+    }
+  }
+}
+```
+
+When adding new tests the CMakeLists.txt file of the respective folder will also need updated to call the test and link the required libraries (e.g. the above example needs the FileParsers library):
+
+```
+rdkit_catch_test(dummyTest catch_dummy.cpp LINK_LIBRARIES FileParsers)
+```
+
+### Documentation
+
+Documentation for the [C++ API](https://www.rdkit.org/docs/cppapi/) is generated from the docstrings, which should be placed above the corresponding function. We'd recommend these take the following format:
+
+```c++
+/*!
+    A brief description of what the function does
+
+    Reference / Source (if there is one)
+
+    \param name: what the parameter is
+    \param name2: what the parameter is
+    etc...
+  */
+```
+
+Where relevant, example useage can also be added to the [Getting Started with C++ Guide](https://www.rdkit.org/docs/GettingStartedInC%2B%2B.html) as part of the pull request.
```

Comment:
please remove this one

Reply:
removed the documentation section and example from the tests, but have left a link to the catch2 framework and where to add new tests (I didn't know this when doing the DCLV stuff!)